pkg/model/gguf_parser_wrapper.go

```diff
@@ -0,0 +1,281 @@
+package model
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	ggufparser "github.com/gpustack/gguf-parser-go"
+)
+
+// Interface for parsing different model formats
+type LocalAIGGUFParser interface {
```

Comment:
Small nit style: it would make the code more reusable if the interface here would require only `ParseGGUFFile` and have different parsers implementing their ParseGGUFFile logic, for instnace a Ollama parser, a Huggingface Parser, etc.

The caller then would need to instantiate the needed parser down the line, for instance:


```go
type GGUFParser interface {
   Parse(path string) (*ggufparser.GGUFFile, error)
}

// GetModelGGufData returns the resources estimation needed to load the model.
func GetModelGGufData(modelPath string, estimator ModelMemoryEstimator, ollamaModel bool) (*ModelEstimate, error) {
	ctx := context.Background()

	fmt.Println("ModelPath: ", modelPath)

        var  ggufParser GGUFParser
 
	// Check if the input is a valid URL
        switch {
          case isURL(modelPath):
             ggufParser = &RemoteFileParser{ctx,modelPath}
           case ollamaModel:
             ggufParser = &OllaamParser{ctx,modelPath}
          /// .. other parsers here
        }
      return estimator.Estimate(ggufRemoteData)
}
```